nil exeption at internal/decoder/slice.go:65

https://drone.gitea.io/go-gitea/gitea/42335/2/15

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x475596]

goroutine 47290 [running]:
testing.tRunner.func1.2(0x24db300, 0x44f29b0)
	/usr/local/go/src/testing/testing.go:1144 +0x332
testing.tRunner.func1(0xc004d59380)
	/usr/local/go/src/testing/testing.go:1147 +0x4b6
panic(0x24db300, 0x44f29b0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
reflect.typedslicecopy(0x2493060, 0xc004fe9e60, 0x2, 0x2, 0x0, 0x2, 0x2, 0xf0)
	/usr/local/go/src/runtime/mbarrier.go:286 +0xaa
github.com/goccy/go-json/internal/decoder.(*sliceDecoder).newSlice(0xc0075fbf10, 0xc0065800a8, 0xf0)
	/drone/src/vendor/github.com/goccy/go-json/internal/decoder/slice.go:65 +0xc5
github.com/goccy/go-json/internal/decoder.(*sliceDecoder).DecodeStream(0xc0075fbf10, 0xc007880a20, 0x3, 0xc0065800a8, 0x9, 0x0)
	/drone/src/vendor/github.com/goccy/go-json/internal/decoder/slice.go:125 +0x3b2
github.com/goccy/go-json/internal/decoder.(*structDecoder).DecodeStream(0xc008fac3c0, 0xc007880a20, 0x2, 0xc006580000, 0xc008438090, 0xc006747570)
	/drone/src/vendor/github.com/goccy/go-json/internal/decoder/struct.go:702 +0x709
github.com/goccy/go-json/internal/decoder.(*ptrDecoder).DecodeStream(0xc0036f2180, 0xc007880a20, 0x1, 0xc006747570, 0x1ff, 0x1ff)
	/drone/src/vendor/github.com/goccy/go-json/internal/decoder/ptr.go:55 +0x96
github.com/goccy/go-json/internal/decoder.(*sliceDecoder).DecodeStream(0xc0075fbf80, 0xc007880a20, 0x1, 0xc003548fa8, 0x0, 0xc003d5a000)
	/drone/src/vendor/github.com/goccy/go-json/internal/decoder/slice.go:149 +0x542
github.com/goccy/go-json.(*Decoder).DecodeWithOption(0xc004fa54f0, 0x237bf20, 0xc003548fa8, 0x0, 0x0, 0x0, 0xc007880a20, 0xc006e88000)
	/drone/src/vendor/github.com/goccy/go-json/decode.go:202 +0x131
github.com/goccy/go-json.(*Decoder).Decode(0xc004fa54f0, 0x237bf20, 0xc003548fa8, 0x2f20480, 0xc004fa54f0)
	/drone/src/vendor/github.com/goccy/go-json/decode.go:168 +0x50
code.gitea.io/gitea/integrations.DecodeJSON(0x2f78ae8, 0xc004d59380, 0xc0024f8c80, 0x237bf20, 0xc003548fa8)
	/drone/src/integrations/integration_test.go:511 +0x92
code.gitea.io/gitea/integrations.TestAPISearchIssues(0xc004d59380)
	/drone/src/integrations/api_issue_test.go:223 +0x1115
testing.tRunner(0xc004d59380, 0x2ad0950)
	/usr/local/go/src/testing/testing.go:1194 +0xef
```

--- a/internal/decoder/slice.go
+++ b/internal/decoder/slice.go
@@ -9,6 +9,13 @@ import (
 	"github.com/goccy/go-json/internal/runtime"
 )
 
+var (
+	sliceType = runtime.Type2RType(
+		reflect.TypeOf((*sliceHeader)(nil)).Elem(),
+	)
+	nilSlice = unsafe.Pointer(&sliceHeader{})
+)
+
 type sliceDecoder struct {
 	elemType          *runtime.Type
 	isElemPointerType bool
@@ -107,7 +114,7 @@ func (d *sliceDecoder) DecodeStream(s *Stream, depth int64, p unsafe.Pointer) er
 			if err := nullBytes(s); err != nil {
 				return err
 			}
-			*(*unsafe.Pointer)(p) = nil
+			typedmemmove(sliceType, p, nilSlice)
 			return nil
 		case '[':
 			s.cursor++
@@ -216,7 +223,7 @@ func (d *sliceDecoder) Decode(ctx *RuntimeContext, cursor, depth int64, p unsafe
 				return 0, err
 			}
 			cursor += 4
-			*(*unsafe.Pointer)(p) = nil
+			typedmemmove(sliceType, p, nilSlice)
 			return cursor, nil
 		case '[':
 			cursor++

